cpp/tests/table/experimental_table_view_tests.cu

```diff
@@ -0,0 +1,280 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/detail/utilities/vector_factories.hpp>
+#include <cudf/table/experimental/row_operators.cuh>
+#include <cudf/table/row_operators.cuh>
+#include <cudf/table/table_device_view.cuh>
+#include <cudf/table/table_view.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/transform.h>
+
+#include <vector>
+
+using namespace cudf::test;
+using namespace cudf::experimental::row;
+
+// Compares two tables row by row, if table1 row is less than table2, then corresponding row value
+// in `output` would be `true`/1 else `false`/0.
+template <typename T>
+struct TypedTableViewTest : public cudf::test::BaseFixture {
+};
+
+using NumericTypesNotBool = Concat<IntegralTypesNotBool, FloatingPointTypes>;
+TYPED_TEST_SUITE(TypedTableViewTest, NumericTypesNotBool);
+
+template <typename Comparator>
+void row_comparison(cudf::table_view input1,
+                    cudf::table_view input2,
+                    cudf::mutable_column_view output,
+                    std::vector<cudf::order> const& column_order)
+{
+  rmm::cuda_stream_view stream{};
+
+  auto table_comparator =
+    lexicographic::two_table_comparator{input1, input2, column_order, {}, stream};
+  auto comparator =
+    table_comparator.device_comparator<cudf::nullate::NO, Comparator>(cudf::nullate::NO{});
+  auto const lhs_it = cudf::experimental::row::lhs_iterator(0);
+  auto const rhs_it = cudf::experimental::row::rhs_iterator(0);
+  thrust::transform(rmm::exec_policy(stream),
+                    lhs_it,
+                    lhs_it + input1.num_rows(),
+                    rhs_it,
+                    output.data<int8_t>(),
+                    comparator);
+}
+
+template <typename Comparator>
+void self_comparison(cudf::table_view input,
+                     cudf::mutable_column_view output,
+                     std::vector<cudf::order> const& column_order)
+{
+  rmm::cuda_stream_view stream{};
+
+  auto table_comparator = lexicographic::self_comparator{input, column_order, {}, stream};
+  auto comparator =
+    table_comparator.device_comparator<cudf::nullate::NO, Comparator>(cudf::nullate::NO{});
+  thrust::transform(rmm::exec_policy(stream),
+                    thrust::make_counting_iterator(0),
+                    thrust::make_counting_iterator(input.num_rows()),
+                    thrust::make_counting_iterator(0),
+                    output.data<int8_t>(),
```

Comment:
What does `bool` do here? I would expect it to map to BOOL8 but I don't know for certain.
```suggestion
                    output.data<bool>(),
```

Reply:
Bool works fine, should have switched it when I copied the original lexicographic test functions